### PR TITLE
Enable collapse on unsigned_long field

### PR DESCRIPTION
**Pull request by mayya-sharipova**
Collapse was not working on unsigned_long field,
as collapsing was enabled only on KeywordFieldType and NumberFieldType.

This introduces a new method `collapseType` to MappedFieldType,
that is checked to decide if collapsing should be enabled.

Relates to #60050

### CODE REVIEW

**Comment by mayya-sharipova** [comment]
@elasticmachine run elasticsearch-ci/packaging-sample-windows

File: server/src/main/java/org/elasticsearch/search/collapse/CollapseBuilder.java [hunk 1]
```diff
@@ -203,12 +202,9 @@ public CollapseContext build(QueryShardContext queryShardContext) {
         if (fieldType == null) {
             throw new IllegalArgumentException("no mapping found for `" + field + "` in order to collapse on");
         }
-        if (fieldType instanceof KeywordFieldMapper.KeywordFieldType == false &&
-            fieldType instanceof NumberFieldMapper.NumberFieldType == false) {
-            throw new IllegalArgumentException("unknown type for collapse field `" + field +
-                "`, only keywords and numbers are accepted");
+        if (fieldType.collapseType() == CollapseType.NONE) {
+            throw new IllegalArgumentException("collapse is not supported on this field type");
```
Comment (cbuescher) [inline server/src/main/java/org/elasticsearch/search/collapse/CollapseBuilder.java:206]: Should we keep adding the field name to the error message to make it more actionable? Maybe we also have the actual field type available here to add to the error?

**Review by cbuescher — COMMENTED** [review]
This looks good, I only left one comment regarding error messages.

File: server/src/main/java/org/elasticsearch/search/collapse/CollapseBuilder.java (hunk 1 above)
Comment (mayya-sharipova) [reply server/src/main/java/org/elasticsearch/search/collapse/CollapseBuilder.java]: @cbuescher Thanks for the feedback, addressed in 621992c

**Review by cbuescher — APPROVED** [review]
Thanks for the changes, LGTM

**Pull request merged**